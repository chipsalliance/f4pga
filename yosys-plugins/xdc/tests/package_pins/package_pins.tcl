yosys -import
if { [info procs get_ports] == {} } { plugin -i design_introspection }
if { [info procs read_xdc] == {} } { plugin -i xdc }
yosys -import  ;# ingest plugin commands

read_verilog $::env(DESIGN_TOP).v
# -flatten is used to ensure that the output eblif has only one module.
# Some of F4PGA expects eblifs with only one module.
synth_xilinx -flatten -abc9 -nosrl -noclkbuf -nodsp

#Read the design constraints
read_xdc -part_json [file dirname [info script]]/../xc7a35tcsg324-1.json $::env(DESIGN_TOP).xdc

# Clean processes before writing JSON.
yosys proc

# Write the design in JSON format.
write_json [test_output_path "package_pins.json"]
