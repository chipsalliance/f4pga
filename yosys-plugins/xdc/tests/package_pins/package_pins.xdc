#OBUF_6
set_property PACKAGE_PIN D10 [get_ports {led[0]}]
set_property DRIVE 12 [get_ports {led[0]}]
#OBUF_7
set_property PACKAGE_PIN A9 [get_ports {led[1]}]
set_property IN_TERM UNTUNED_SPLIT_40 [get_ports led[1]]
set_property SLEW FAST [get_ports led[1]]
set_property IOSTANDARD SSTL135 [get_ports led[1]]
#OBUF_OUT
set_property PACKAGE_PIN E3 [get_ports out_a]
set_property IN_TERM UNTUNED_SPLIT_50 [get_ports out_a]
set_property SLEW FAST [get_ports out_a]
set_property IOSTANDARD LVCMOS33 [get_ports out_a]
#bottom_inst.OBUF_10
set_property PACKAGE_PIN C2 [get_ports {out_b[0]}]
set_property SLEW SLOW [get_ports {out_b[0]}]
set_property IOSTANDARD LVCMOS18 [get_ports {out_b[0]}]
#bottom_inst.OBUF_11
set_property PACKAGE_PIN R2 [get_ports {out_b[1]}]
set_property DRIVE 4 [get_ports {out_b[1]}]
set_property SLEW FAST [get_ports {out_b[1]}]
set_property IOSTANDARD LVCMOS25 [get_ports {out_b[1]}]
#bottom_inst.OBUF_9
set_property PACKAGE_PIN M6 [get_ports {led[2]}]
set_property SLEW FAST [get_ports {led[2]}]
set_property IOSTANDARD DIFF_SSTL135 [get_ports {led[2]}]
#bottom_intermediate_inst.OBUF_8
set_property PACKAGE_PIN N4 [get_ports {led[3]}]
set_property DRIVE 16 [get_ports {led[3]}]
set_property IOSTANDARD SSTL135 [get_ports {led[3]}]
#OBUFTDS_2
set_property PACKAGE_PIN N2 [get_ports signal_p]
set_property PACKAGE_PIN N1 [get_ports signal_n]
set_property SLEW FAST [get_ports signal_p]
set_property IOSTANDARD DIFF_SSTL135 [get_ports signal_p]

