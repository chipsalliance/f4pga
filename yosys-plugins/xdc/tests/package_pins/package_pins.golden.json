{
  "OBUFTDS_2": {
    "IOSTANDARD": "DIFF_SSTL135",
    "IO_LOC_PAIRS": "signal_p:N2,signal_n:N1",
    "SLEW": "FAST"
  },
  "OBUF_6": {
    "DRIVE": "12",
    "IOSTANDARD": "LVCMOS33",
    "IO_LOC_PAIRS": "led[0]:D10",
    "SLEW": "SLOW"
  },
  "OBUF_7": {
    "IN_TERM": "UNTUNED_SPLIT_40",
    "IOSTANDARD": "SSTL135",
    "IO_LOC_PAIRS": "led[1]:A9",
    "SLEW": "FAST"
  },
  "OBUF_OUT": {
    "IN_TERM": "UNTUNED_SPLIT_50",
    "IOSTANDARD": "LVCMOS33",
    "IO_LOC_PAIRS": "out_a:E3",
    "SLEW": "FAST"
  },
  "bottom_inst.OBUF_10": {
    "IOSTANDARD": "LVCMOS18",
    "IO_LOC_PAIRS": "out_b[0]:C2",
    "SLEW": "SLOW"
  },
  "bottom_inst.OBUF_11": {
    "DRIVE": "4",
    "IOSTANDARD": "LVCMOS25",
    "IO_LOC_PAIRS": "out_b[1]:R2",
    "SLEW": "FAST"
  },
  "bottom_inst.OBUF_9": {
    "IOSTANDARD": "DIFF_SSTL135",
    "IO_LOC_PAIRS": "led[2]:M6",
    "SLEW": "FAST"
  },
  "bottom_intermediate_inst.OBUF_8": {
    "DRIVE": "16",
    "IOSTANDARD": "SSTL135",
    "IO_LOC_PAIRS": "led[3]:N4",
    "SLEW": "SLOW"
  }
}